Aligning the front-end card with latest Plant component changes

PeteBa opened the pull request:
This PR continues work-in-progress from PR #583 that was merged before finalising.

This PR aligns the unit-of-measurement labels with those provided by the backend plant component but is still dependent on an outstanding review comment on the backend plant component [PR9524](https://github.com/home-assistant/home-assistant/pull/9534).

PeteBa committed:
Aligning with latest backend plant component changes

balloob commented on src/cards/ha-plant-card.html:
```diff
@@ -109,7 +109,10 @@
 
     computeUOM(stateObj, sensor) {
-      return _CONFIG[sensor].uom;
+      if (typeof stateObj.attributes.unit_of_measurement[sensor] === 'undefined') {
```
This is going to be renamed. `unit_of_measurement` should never be an object.

balloob replied:
btw, you can check for existence of a property with `if (sensor in stateObj.attributes.units)`

PeteBa replied:
I'm a convert to "in" syntax for attribute checking! Much more readable. Both comments addressed in latest commit.

ChristianKuehnel commented:
I just renamed the attributes in the plant-card as I did in the python code, but I seem to be too stupid to create a pull request ontop of your branch for this. Anyway it's only a few changes: https://github.com/ChristianKuehnel/home-assistant-polymer/commit/9c20de4185059c77408c2ad1f0688c16542e8e91

What I notices with my changes now is  that a click on the sun icon (=brightness) does not show the history any more. This is not surprising as there is not history for max_brightness. Is there a way to display the brightness history instead of the max_brightness history?

PeteBa committed:
Address review comments and plant component alignment

PeteBa commented:
@ChristianKuehnel , not sure that others can add to an in-progress PR but really appreciate if you could check out the latest commit and confirm it aligns with your back-end changes. I end up needing a "special case" for max_brightness. I use the icon/uom/problem/sensor information for "brightness" but swap in "max_brightness" for the value if it is available. So if recorder has not been enabled the user will get real-time brightness otherwise they get the more informative historical max. I think this is a good compromise without adding more icons and two brightness measures. Happy to take feedback on this.

balloob commented:
Heads up: I've disabled the plant card in the latest master so I could cut a release of the frontend for 0.58.

ChristianKuehnel commented:
@PeteBa I testet the latest version and do not have any complaints :+1: 

The brightness history is now showing the brightness history :wink: 
From my point of view this can be merged.

PeteBa committed:
Revert PR636 to reinstate card

PeteBa commented:
I have reverted the PR #636 changes to reinstate the card and removed WIP from the PR title.

I believe this can be merged once the back-end PR [#9534](https://github.com/home-assistant/home-assistant/pull/9534) has been approved.

PeteBa commented:
The associated back-end PR has an intermittent issue that is proving difficult to resolve. I will close this to avoid clutter and re-open once there is a resolution to [#9534](https://github.com/home-assistant/home-assistant/pull/9534).

PeteBa committed:
Add plant domain icon

homeassistant commented:
Hello @PeteBa,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`0889da4bfe1712b35f2aa44ec72edd7ab91bfe15`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant-polymer&#x2F;pull&#x2F;633&#x2F;commits&#x2F;0889da4bfe1712b35f2aa44ec72edd7ab91bfe15) - This commit has something that looks like an email address (pete_ba@yahoo.co.uk). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

PeteBa commented:
This PR is ready for review/approval. The back-end dependency on [#9534](https://github.com/home-assistant/home-assistant/pull/9534) has been closed out.
